Create a whitelist of bulbs

Djelibeybi opened the pull request:
Now that the newer LIFX bulbs are HomeKit enabled, it may be more convenient to list the bulbs you want to control via Homebridge instead of creating a list of all the ones you don't. This provides both  whitelist (`enabledDevices`) and blacklist (`ignoredDevices`) functionality.

Djelibeybi committed:
Adding an enabledDevices array so that it's possible to whitelist
bulb MAC addresses instead of listing all the ones to ignore.

Signed-off-by: Avi Miller <me@dje.li>

Djelibeybi committed:
Adding an enabledDevices array so that it's possible to whitelist
bulb MAC addresses instead of listing all the ones to ignore.

Signed-off-by: Avi Miller <me@dje.li>

devbobo commented on index.js:
```diff
@@ -96,7 +100,12 @@ function LifxLanPlatform(log, config, api) {
+    }
+
     this.ignoredDevices = this.config.ignoredDevices || [];
+    this.enabledDevices = this.config.enabledDevices || [];
```
I think this line needs to be...
```this.enabledDevices = this.config.enabledDevices || null;```

devbobo commented on index.js:
```diff
@@ -124,6 +133,14 @@ function LifxLanPlatform(log, config, api) {
 
             return;
         }
+        else if (this.enabledDevices.indexOf(bulb.id) === -1) {
```
along with my requested change at line 108, I think this line needs to be
```else if (this.enabledDevices && this.enabledDevices.indexOf(bulb.id) === -1) {```

otherwise, `this.enabledDevices` will always be an empty array when not defined in the `config.json` (ie: current operation) and all devices will be removed.

devbobo commented on index.js:
```diff
@@ -146,6 +163,14 @@ function LifxLanPlatform(log, config, api) {
 
             return;
         }
+        else if (this.enabledDevices.indexOf(bulb.id) === -1) {
```
this line also needs to be...
```else if (this.enabledDevices && this.enabledDevices.indexOf(bulb.id) === -1) {```

devbobo reviewed (changes requested):
Thanks for implementing this...can you please look at my comments. I think the code as it currently stands will remove all accessories for any user who doesn't have `enabledDevices` in their `config.json`

Djelibeybi commented:
Thanks, I have updated accordingly. This is my first node.js pull request, so the feedback is welcome! :)

I've also rebased to master so that the merge conflict is resolved.

Djelibeybi commented:
Ping @devbobo. Any chance of a fresh review? I've squashed it all down to a single commit too.

Commit:
Update README.md

DerAndereAndi commented:
It would be awesome to see this being added to an update

Djelibeybi committed:
Merge branch 'whitelist' of github.com:Djelibeybi/homebridge-lifx-lan into whitelist
